core/src/main/java/io/crate/core/collections/StringObjectMaps.java

```diff
@@ -22,45 +22,35 @@
 package io.crate.core.collections;
 
 import io.crate.core.StringUtils;
-import org.elasticsearch.common.Nullable;
 
+import javax.annotation.Nullable;
 import java.util.List;
 import java.util.Map;
+import java.util.RandomAccess;
 
-public class StringObjectMaps {
+public final class StringObjectMaps {
 
     @Nullable
     public static Object getByPath(Map<String, Object> map, String path) {
         assert path != null : "path should not be null";
-        Object tmp;
-        List<String> splittedPath = StringUtils.PATH_SPLITTER.splitToList(path);
-        for (String pathElement : splittedPath.subList(0, splittedPath.size() - 1)) {
-            tmp = map.get(pathElement);
-            if (tmp != null && tmp instanceof Map) {
-                map = (Map<String, Object>) tmp;
-            } else {
-                break;
-            }
-        }
-
-        if (map != null) {
-            // get last path element
-            return map.get(splittedPath.get(splittedPath.size() - 1));
-        }
-        return null;
+        return getByPath(map, StringUtils.PATH_SPLITTER.splitToList(path));
     }
 
-    public static Object fromMapByPath(Map value, List<String> path) {
+    @Nullable
+    public static Object getByPath(Map value, List<String> path) {
```

Comment:
would it make sense for this map to be typed as well ( <String, Object> ) ?

Reply:
@andreidan yep, pushed a fixup